app_dart/lib/src/service/agent_service.dart

```diff
@@ -0,0 +1,41 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:math';
+
+import 'package:dbcrypt/dbcrypt.dart';
+import 'package:meta/meta.dart';
+
+/// Service class for Agent.
+///
+/// This service exists to provide common agent queries made by
+/// the Cocoon backend.
+@immutable
+class AgentService {
```

Comment:
You should probably add a const constructor (since you can).

Reply:
Added.